sourcemap/mapper.go

```diff
@@ -66,3 +81,40 @@ func abs(n int) int {
 	}
 	return n
 }
+
+// limitScanLines is a modified form of bufio.ScanLines. It truncates any lines
+// longer than the supplied `max` value.
+func limitScanLines(max int) func([]byte, bool) (int, []byte, error) {
```

Comment:
This is mostly copy and paste. I'm happy to remove this and just re-slice `scanner.Text()` above, but i figured if we're trying to avoid allocations, why not try to avoid allocations. (`scanner.Text()` allocates a new string)

Reply:
Yeah, it won't decrease the number of allocations, but I guess they'd potentially be smaller (which probably isn't much of a win).